src/control/BarrettFingerKinematicSimulationPositionCommandExecutor.cpp

```diff
@@ -72,47 +96,47 @@ std::future<void> BarrettFingerPositionCommandExecutor::execute(
       throw std::runtime_error("Another command in execution.");
 
     mPromise.reset(new std::promise<void>());
-    mCollideWith = _collideWith;
     mInExecution = true;
     mDistalOnly = false;
 
     // Set mProximalGoalPosition.
-    if (_goalPosition < mProximalLimits.first)
+    if (goalPosition < mProximalLimits.first)
       mProximalGoalPosition = mProximalLimits.first;
-    else if (_goalPosition > mProximalLimits.second)
+    else if (goalPosition > mProximalLimits.second)
       mProximalGoalPosition = mProximalLimits.second;
     else
-      mProximalGoalPosition = _goalPosition;
+      mProximalGoalPosition = goalPosition;
 
     return mPromise->get_future();
   }
 }
 
 //=============================================================================
-void BarrettFingerPositionCommandExecutor::terminate()
+void BarrettFingerKinematicSimulationPositionCommandExecutor::terminate()
 {
   mPromise->set_value();
   mInExecution = false;
-  mCollideWith.reset();
 }
+
 //=============================================================================
-void BarrettFingerPositionCommandExecutor::step(double _timeSincePreviousCall)
+void BarrettFingerKinematicSimulationPositionCommandExecutor::step(
+ const std::chrono::milliseconds& timeSincePreviousCall)
```

Comment:
I was thinking that we'd have a `mTimeOfPreviousCall` member variable that gets set to `now()` at the bottom of each function call. Then, we'd use `now() - mTimeOfPreviousCall` as the time since the previous call.